### ISSUE TITLE
Target symbol-name is unknown: 'loki' with NLog 5

### ISSUE THREAD

**Issue by maradev21** [corentinaltepe/nlog.loki#34]
I tried to use the library to write my logs to Loki (installed as docker in a linux virtual machine on my pc, it's just for testing purposes), but it seems no log is written there.

I tried to enable NLog's internal logger, and this is what I found in the file:

```
2022-05-23 15:48:13.7420 Info Message Template Auto Format enabled
2022-05-23 15:48:13.7420 Info Loading assembly: NLog.Loki
2022-05-23 15:48:13.8115 Warn Loaded 21 valid types from Assembly: NLog.Loki, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null Exception: System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
Declaration referenced in a method implementation cannot be a final method.  Type: 'NLog.Loki.LokiTarget'.  Assembly: 'NLog.Loki, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.Assembly.GetTypes()
   at NLog.Internal.ReflectionHelpers.SafeGetTypes(Assembly assembly)
System.TypeLoadException: Declaration referenced in a method implementation cannot be a final method.  Type: 'NLog.Loki.LokiTarget'.  Assembly: 'NLog.Loki, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
2022-05-23 15:48:13.8282 Warn Type load exception. Exception: System.TypeLoadException: Declaration referenced in a method implementation cannot be a final method.  Type: 'NLog.Loki.LokiTarget'.  Assembly: 'NLog.Loki, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.

...

2022-05-23 15:48:13.8548 Warn Error has been raised. Exception: NLog.NLogConfigurationException: Failed to create Target of type: loki
 ---> System.ArgumentException: Target symbol-name is unknown: 'loki'
   at NLog.Config.Factory`2.CreateInstance(String itemName)
   at NLog.Config.LoggingConfigurationParser.FactoryCreateInstance[T](String classType, INamedItemFactory`2 factory)
   --- End of inner exception stack trace ---
2022-05-23 15:48:13.8648 Warn Error has been raised. Exception: NLog.NLogConfigurationException: Target 'lokiLogger' not found for logging rule: *.
2022-05-23 15:48:13.8648 Info Validating config: TargetNames=console, ConfigItems=8, FilePath=E:\Users\zorattim\Desktop\ProvaQuickGraph\LogTests\bin\Debug\net6.0\NLog.config
2022-05-23 15:48:13.8976 Info Configuration initialized.
2022-05-23 15:48:13.8976 Info NLog, Version=5.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c. File version: 5.0.0.1032. Product version: 5.0.0+2e132d979024a14ffe7c91aa3615f720b8ba6be1. GlobalAssemblyCache: False
2022-05-23 15:48:18.3589 Info AppDomain Shutting down. LogFactory closing...
2022-05-23 15:48:18.3746 Info LogFactory has been closed.
```

This is my `NLog.config` file:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="false"
      internalLogLevel="Info" internalLogFile="${BaseDir}/Log/nlog-internal.log">
	<variable name="TestLogLayout"           value="[${level}] ${date} - ${message} ${all-event-properties}"/>

	<extensions>
		<add assembly="NLog.Loki" />
	</extensions>

	<targets async="true">
		<target name="console" xsi:type="ColoredConsole" header="${gdc:item=LogHeader}" layout="${var:TestLogLayout}"/>

		<target name="lokiLogger"
				xsi:type="loki"
				endpoint="http://<ip>:3100"
				orderWrites="true"
				compressionLevel="noCompression"
				layout="${var:TestLogLayout}">
			<label name="app" layout="nlog-tests" />
			<label name="server" layout="${hostname:lowercase=true}" />
		</target>
	</targets>
	
	<rules>
		<logger name="*" minlevel="Trace"  writeTo="console" />
		<logger name="*" minlevel="Trace" writeTo="lokiLogger"/>
	</rules>
</nlog>
```

The issue seems being the usage of **NLog 5**: if I remove that reference and I just use **NLog 4.7.15** (referenced by NLog.Targets.Loki) I find my logs on Loki and everything seems working.

Am I missing something? Maybe in the latest release of NLog there are some changes that are incompatible with this package?

PS: It's the first time I open an issue on github, so please, don't hate me if I wrote something wrong 🙏🏻

**Comment by corentinaltepe** [comment]
Hello. Thank you for all the information. There must be breaking change in the new version of NLog, or maybe I need to release a new version referencing NLog v5. I'll see if I can reproduce the issue and fix it. Any contribution is welcome if you find a fix in the meantime.

**Comment by corentinaltepe** [comment]
Hello @maradev21. Thanks again for reporting the issue. It turns out NLog.Targets.Loki was not compatible with NLog v5.

I have reproduced it and I believe I have solved it in #35. Could you please upgrade your dependency to [v1.4.2](https://www.nuget.org/packages/NLog.Targets.Loki/1.4.2) (just released now) and let me know if that new package solves your issue?

**Comment by maradev21** [comment]
Hi @corentinaltepe, I confirm that with the new version it works. Thank you
